### ISSUE TITLE
Construction of nested pydantic models using plain dictionaries

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#3510]
Pydantic allows construction of [nested models](https://pydantic.dev/docs/validation/latest/concepts/models/#nested-models) using a dictionary:

```py
from pydantic import BaseModel

class Child(BaseModel):
    value: int

class Parent(BaseModel):
    child: Child

Parent(child={"value": 1})
```

In ty (and pyright, and mypy), this leads to *"Expected `Child`, found `dict[str, int]`"*. Pyrefly apparently "supports" this by giving the `Parent.__init__` a signature of `(self: Parent, *, child: Any, **Unknown) -> None` in [lax mode](https://pyrefly.org/en/docs/pydantic/#validation-modes), i.e. there is no validation at all. In strict mode, it also fails to type check this.

This seems relatively complex to support. I guess we could synthesize a TypedDict for a given model and then allow `child` in the constructor of `Parent` to be of type `Child | ChildTypedDict`?

**Comment by sharkdp** [comment]
We now also "support" this in lax mode in that we don't raise false positives, since we also map `Child -> Any`. I'm closing this for now, but feel free to comment here if anyone thinks we should improve our support here.

**Comment by sharkdp** [comment]
Reconsidering this after the discussion in https://github.com/astral-sh/ty/issues/3947
